pkg/controller/secret/secretsync.go

```diff
@@ -182,6 +199,27 @@ func (c *SecretSyncController) translateSecret(cloudSecret *v1.Secret, cloudConf
 	return &secret, nil
 }
 
+// fetchEndpoint fetches the url provided against endpointkey
+func fetchEndpoint(endpointkey, conf, defaultEndpointValue string) (string, error) {
+	var re *regexp.Regexp
+	var match []string
+
+	// Fetch url provided against endpointkey
+	re = regexp.MustCompile(fmt.Sprintf("%s = (.*?)\n", endpointkey))
+	match = re.FindStringSubmatch(conf)
+	// If the url doesn't exist return default value
+	if len(match) <= 1 {
+		return defaultEndpointValue, nil
+	}
+
+	// If the url exists, validate it
+	if _, err := url.ParseRequestURI(match[1]); err != nil {
```

Comment:
Can we use _, err := url.Parse(urlString) method instead parse URI, may be check which one will be more appropriate

Reply:
**url.ParseRequestURI** is a better choice that **url.Parse(urlString)**, reason being url.Parse accepts inputs like 
1. "https//:3003"
2. "/iam/endpoint", etc

But url.ParseRequestURI identifies these,

The difference is documented here. If we do not want stricter validation from the the operator, I can switch to parse.URL, I would request @cjschaef comment also on this.
https://pkg.go.dev/net/url#Parse
https://pkg.go.dev/net/url#ParseRequestURI

Reply:
The URL's are being strictly parsed currently by the installer code. But if you'd prefer additional validation that is up to you.

For instance, it occurs in two locations, when loading an `install-config` and later when creating cluster manifests
https://github.com/openshift/installer/pull/7632/files#diff-e6306df593d77282680d1864f5e1504e044abdbe681195fbcbeaab0cc3c45cc2R454-R467

https://github.com/openshift/installer/pull/7632/files#diff-da80c878bba406f82e10ab78858cbaff8bb20a72f7aed6ce10051c4ab550441fR99-R130

Reply:
Thank you @cjschaef . Will discuss and update the code.

Reply:
@arahamad sir, url.Parse also accepts an empty string. We can discuss and conclude.